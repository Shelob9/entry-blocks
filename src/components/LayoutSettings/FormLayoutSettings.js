import {RenderGroup} from '@caldera-labs/components';
import React from "react";

function createConfigFields(settings,layouts,onChange) {
	const {
		layout,
		pdf_layout,
		pdf_link
	} = settings;

	const ID = 'caldera-forms-pro-layout-settings';

	/**

	 */

	const layoutOpts = [
		{
			label: '--',
			value: null
		}
	];
	Object.values(layouts).forEach(layout => {
		layoutOpts.push({
			label: layout.title,
			value: layout.ID,
		});
	});

	return [
		{
			id: ID + '-layout',
			label: 'Layout',
			desc: 'Main Message Layout',
			type: 'select',
			value: layout,
			options: layoutOpts,
			onValueChange: (layout) => {
				onChange({
					...settings,
					layout
				})
			}
		},
		{
			id: ID + '-pdf_layout',
			label: 'PDF Layout',
			desc: 'PDF Layout',
			type: 'select',
			value: pdf_layout,
			options: layoutOpts,
			onValueChange: (pdf_layout) => {
				onChange({
					...settings,
					pdf_layout
				})
			}
		},
		{
			id: ID + '-pdf_link',
			label: 'PDF Link',
			desc: 'Main Message Layout',
			type: 'checkbox',
			value: pdf_link,
			options: [
				{
					value: true,
					label: 'Enable'
				}
			],
			onValueChange: (pdf_link) => {
				onChange({
					...settings,
					pdf_link
				})
			}
		},

	];


}

export const FormLayoutSettings = ({settings,layouts,onChange}) => {
	return (
		<RenderGroup
			configFields={createConfigFields(settings,layouts,onChange)}
			className={'cf-mail-form-layout-settings'}
		/>
	)
}
