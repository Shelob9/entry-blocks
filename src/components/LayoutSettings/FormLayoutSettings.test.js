import React from "react";
import renderer from "react-test-renderer";
import {FormLayoutSettings} from "./FormLayoutSettings";
describe( 'Form layout settings', () => {
	const settings = {
		attach_pdf: false,
		pdf_link: 0,
		layout: 0,
		pdf_layout: 0,
		send_local: false,
		form_id: "CF5be77c7b45877",
		name: "Contact Form 1"
	};

	const layouts = [
		{
			ID: 2,
			title: 'Layout 2',
		},
		{
			ID: 3,
			title: 'Layout 3',
		}
	];

	it( 'matches snapshot', () => {
		expect(
			renderer.create(
				<FormLayoutSettings
					layouts={layouts}
					settings={settings}
					onChange={jest.fn()}
				/>

			).toJSON()
		).toMatchSnapshot()
	});
});
