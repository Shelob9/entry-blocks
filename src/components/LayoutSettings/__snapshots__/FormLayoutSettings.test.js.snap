exports[`Form layout settings matches snapshot 1`] = `
<div
  className="cf-mail-form-layout-settings">
  <div
    className="caldera-config-group">
    <div
      className="caldera-config-field">
      <label
        htmlFor="caldera-forms-pro-layout-settings-layout">
        Layout
      </label>
      <select
        className="field-config field-config"
        disabled={false}
        id="caldera-forms-pro-layout-settings-layout"
        onChange={[Function]}
        value={0}>
        <option
          className="caldera-config-option"
          value={null}>
          --
        </option>
        <option
          className="caldera-config-option"
          value={2}>
          Layout 2
        </option>
        <option
          className="caldera-config-option"
          value={3}>
          Layout 3
        </option>
      </select>
      <p
        className="description"
        id="caldera-forms-pro-layout-settings-layout-description">
        Main Message Layout
      </p>
    </div>
  </div>
  <div
    className="caldera-config-group">
    <div
      className="caldera-config-field">
      <label
        htmlFor="caldera-forms-pro-layout-settings-pdf_layout">
        PDF Layout
      </label>
      <select
        className="field-config field-config"
        disabled={false}
        id="caldera-forms-pro-layout-settings-pdf_layout"
        onChange={[Function]}
        value={0}>
        <option
          className="caldera-config-option"
          value={null}>
          --
        </option>
        <option
          className="caldera-config-option"
          value={2}>
          Layout 2
        </option>
        <option
          className="caldera-config-option"
          value={3}>
          Layout 3
        </option>
      </select>
      <p
        className="description"
        id="caldera-forms-pro-layout-settings-pdf_layout-description">
        PDF Layout
      </p>
    </div>
  </div>
  <div
    className="caldera-config-group">
    <div
      className="caldera-config-field">
      <label
        htmlFor="caldera-forms-pro-layout-settings-pdf_link">
        PDF Link
      </label>
      <input
        aria-describedby="caldera-forms-pro-layout-settings-pdf_link-description"
        className="field-config field-config"
        defaultChecked={0}
        disabled={false}
        id="caldera-forms-pro-layout-settings-pdf_link"
        onBlur={undefined}
        onChange={[Function]}
        onFocus={undefined}
        required={false}
        type="checkbox" />
      <p
        className="description"
        id="caldera-forms-pro-layout-settings-pdf_link-description">
        Main Message Layout
      </p>
    </div>
  </div>
</div>
`;
